File: assignment 9/python/programs/9.py
def find_min_repeating_index(arr):
    element_indices = {}  # Dictionary to store the indices of elements
    min_repeating_index = float("inf")

    for index, element in enumerate(arr):
        if element in element_indices:
            min_repeating_index = min(min_repeating_index, element_indices[element])
        else:
            element_indices[element] = index

    if min_repeating_index == float("inf"):
        return -1  # No repeating element found
    else:
        return min_repeating_index


# Example usage
arr = [4, 2, 6, 1, 6, 5, 8, 2]
min_index = find_min_repeating_index(arr)
if min_index == -1:
    print("No repeating element found.")
else:
    print("Minimum index of a repeating element:", min_index)

File: assignment 9/c/programs/9.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void findMinimumIndexOfRepeatingElement(int arr[], int size)
{
    // Create a dictionary to store the index of the first occurrence of each element
    int *firstOccurrence = (int *)malloc(size * sizeof(int));
    for (int i = 0; i < size; i++)
    {
        firstOccurrence[i] = -1; // Initialize with -1 to indicate no occurrence yet
    }

    int minIndex = size;

    for (int i = 0; i < size; i++)
    {
        if (firstOccurrence[arr[i]] == -1)
        {
            // This element is seen for the first time
            firstOccurrence[arr[i]] = i;
        }
        else
        {
            // This element has been seen before, update minIndex if needed
            if (firstOccurrence[arr[i]] < minIndex)
            {
                minIndex = firstOccurrence[arr[i]];
            }
        }
    }

    if (minIndex < size)
    {
        printf("Minimum index of a repeating element: %d\n", minIndex);
    }
    else
    {
        printf("No repeating elements found.\n");
    }

    free(firstOccurrence);
}

int main()
{
    int arr[] = {4, 2, 6, 1, 6, 2, 4};
    int size = sizeof(arr) / sizeof(arr[0]);

    findMinimumIndexOfRepeatingElement(arr, size);

    return 0;
}
